enterprise/cluster/src/main/java/org/neo4j/cluster/com/NetworkSender.java

```diff
@@ -424,6 +426,8 @@ public ChannelPipeline getPipeline() throws Exception
     private class NetworkMessageSender
             extends SimpleChannelHandler
     {
+        private Throwable lastException;
```

Comment:
I assume it might be the right place to use `org.neo4j.kernel.impl.util.CappedOperation`. It would also require some tweaks to be thread safe as it looks like `exceptionCaught` and `writeComplete` might be called by different threads.